Adding unix permissions

aanm opened the pull request:
Fixes #300
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/cilium/cilium/pull/356%23discussion_r106825373%22%2C%20%22https%3A//github.com/cilium/cilium/pull/356%23discussion_r106825416%22%2C%20%22https%3A//github.com/cilium/cilium/pull/356%23discussion_r106825482%22%2C%20%22https%3A//github.com/cilium/cilium/pull/356%23discussion_r106825519%22%2C%20%22https%3A//github.com/cilium/cilium/pull/356%23discussion_r106825568%22%2C%20%22https%3A//github.com/cilium/cilium/pull/356%23discussion_r106825583%22%2C%20%22https%3A//github.com/cilium/cilium/pull/356%23discussion_r106826251%22%2C%20%22https%3A//github.com/cilium/cilium/pull/356%23discussion_r107158677%22%5D%2C%20%22comments%22%3A%20%7B%22Pull%208c6e2329fd4b7d7846b0f73acfdc285106a449ae%20daemon/defaults/defaults.go%205%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/cilium/cilium/pull/356%23discussion_r106825373%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Why%20do%20you%20want%20this%20to%20be%20world%20readable%3F%22%2C%20%22created_at%22%3A%20%222017-03-20T01%3A07%3A16Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars2.githubusercontent.com/u/1417913%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/tgraf%22%7D%7D%2C%20%7B%22body%22%3A%20%22Otherwise%20I%20have%20to%20change%20the%20directory%20group%20to%20belong%20to%20%60ciliium%60%20as%20well.%5Cr%5Cn%5Cr%5CnI%20couldn%27t%20reach%20the%20cilium%20socket%20inside%20without%20%60775%60%20%22%2C%20%22created_at%22%3A%20%222017-03-20T01%3A11%3A38Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars3.githubusercontent.com/u/5714066%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/aanm%22%7D%7D%2C%20%7B%22body%22%3A%20%22Changing%20the%20directory%20group%20ownership%20sounds%20much%20better%20to%20me%20than%20making%20the%20entire%20directory%20world%20readable.%22%2C%20%22created_at%22%3A%20%222017-03-20T01%3A12%3A44Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars2.githubusercontent.com/u/1417913%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/tgraf%22%7D%7D%2C%20%7B%22body%22%3A%20%22Should%20we%20change%20the%20cilium%20socket%20to%20%60/var/run/cilium.sock%60%20instead%20of%20%60/var/run/cilium/cilium.sock%60%3F%22%2C%20%22created_at%22%3A%20%222017-03-20T01%3A14%3A30Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars3.githubusercontent.com/u/5714066%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/aanm%22%7D%7D%2C%20%7B%22body%22%3A%20%22Yes%2C%20let%27s%20do%20that.%20That%20we%20can%20keep%20the%20runtime%20directory%20safe.%22%2C%20%22created_at%22%3A%20%222017-03-20T01%3A35%3A22Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars2.githubusercontent.com/u/1417913%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/tgraf%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20daemon/defaults/defaults.go%3AL19-26%22%7D%2C%20%22Pull%203eaa73f9b201e193d62bf00edfab7c1eac80cbb5%20Makefile%205%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/cilium/cilium/pull/356%23discussion_r107158677%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22I%27m%20fine%20disabling%20it%20for%20now%20until%20we%20have%20the%20options%20in%20place%20to%20support%20a%20certificate.%22%2C%20%22created_at%22%3A%20%222017-03-21T13%3A51%3A12Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars2.githubusercontent.com/u/1417913%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/tgraf%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20Makefile%3AL85-92%22%7D%2C%20%22Pull%200214a7374c9578e45b643caded9cc14fafaab6e1%20api/v1/server/server.go%2053%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/cilium/cilium/pull/356%23discussion_r106825416%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22I%20think%20this%20code%20will%20be%20overwritten%20when%20we%20call%2C%20%60make%20generate-api%60%2C%20no%3F%22%2C%20%22created_at%22%3A%20%222017-03-20T01%3A08%3A41Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars2.githubusercontent.com/u/1417913%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/tgraf%22%7D%7D%2C%20%7B%22body%22%3A%20%22argh..%20you%27re%20right...%20%3A%28%20let%20me%20see%20if%20I%20can%20workaround%20this%22%2C%20%22created_at%22%3A%20%222017-03-20T01%3A15%3A10Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars3.githubusercontent.com/u/5714066%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/aanm%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20api/v1/server/server.go%3AL170-193%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'><img src='http://www.codereviewhub.com/site/github-bar.png' height=40></a>
- [x] <a href='#crh-comment-Pull 8c6e2329fd4b7d7846b0f73acfdc285106a449ae daemon/defaults/defaults.go 5'></a> <img src='http://www.codereviewhub.com/site/github-remaining.png' height=16 width=60>&nbsp;<b><a href='https://github.com/cilium/cilium/pull/356#discussion_r106825373'>File: daemon/defaults/defaults.go:L19-26</a></b>
- <a href='https://github.com/tgraf'><img border=0 src='https://avatars2.githubusercontent.com/u/1417913?v=3' height=16 width=16></a> Why do you want this to be world readable?
- <a href='https://github.com/aanm'><img border=0 src='https://avatars3.githubusercontent.com/u/5714066?v=3' height=16 width=16></a> Otherwise I have to change the directory group to belong to `ciliium` as well.
I couldn't reach the cilium socket inside without `775`
- <a href='https://github.com/tgraf'><img border=0 src='https://avatars2.githubusercontent.com/u/1417913?v=3' height=16 width=16></a> Changing the directory group ownership sounds much better to me than making the entire directory world readable.
- <a href='https://github.com/aanm'><img border=0 src='https://avatars3.githubusercontent.com/u/5714066?v=3' height=16 width=16></a> Should we change the cilium socket to `/var/run/cilium.sock` instead of `/var/run/cilium/cilium.sock`?
- <a href='https://github.com/tgraf'><img border=0 src='https://avatars2.githubusercontent.com/u/1417913?v=3' height=16 width=16></a> Yes, let's do that. That we can keep the runtime directory safe.
- [x] <a href='#crh-comment-Pull 0214a7374c9578e45b643caded9cc14fafaab6e1 api/v1/server/server.go 53'></a> <img src='http://www.codereviewhub.com/site/github-remaining.png' height=16 width=60>&nbsp;<b><a href='https://github.com/cilium/cilium/pull/356#discussion_r106825416'>File: api/v1/server/server.go:L170-193</a></b>
- <a href='https://github.com/tgraf'><img border=0 src='https://avatars2.githubusercontent.com/u/1417913?v=3' height=16 width=16></a> I think this code will be overwritten when we call, `make generate-api`, no?
- <a href='https://github.com/aanm'><img border=0 src='https://avatars3.githubusercontent.com/u/5714066?v=3' height=16 width=16></a> argh.. you're right... :( let me see if I can workaround this
- [x] <a href='#crh-comment-Pull 3eaa73f9b201e193d62bf00edfab7c1eac80cbb5 Makefile 5'></a> <img src='http://www.codereviewhub.com/site/github-remaining.png' height=16 width=60>&nbsp;<b><a href='https://github.com/cilium/cilium/pull/356#discussion_r107158677'>File: Makefile:L85-92</a></b>
- <a href='https://github.com/tgraf'><img border=0 src='https://avatars2.githubusercontent.com/u/1417913?v=3' height=16 width=16></a> I'm fine disabling it for now until we have the options in place to support a certificate.


<a href='https://www.codereviewhub.com/cilium/cilium/pull/356?mark_as_completed=1'><img src='http://www.codereviewhub.com/site/github-mark-as-completed.png' height=26></a>&nbsp;<a href='https://www.codereviewhub.com/cilium/cilium/pull/356?approve=1'><img src='http://www.codereviewhub.com/site/github-approve.png' height=26></a>&nbsp;<a href='https://github.com/cilium/cilium/pull/356'><img src='http://www.codereviewhub.com/site/github-refresh.png' height=26></a>
<a href='#crh-end'></a>

aanm committed:
defaults: changed cilium socket path to /var/run/cilium.sock

Signed-off-by: André Martins <andre@cilium.io>

aanm committed:
server: changed cilium socket permissions

Changed cilium socket to user.group root.cilium and permissions to 660.
This way users in cilium group will be able to reach the daemon without
the need to use root privileges.

While regenerating the api, some models where cleaned up.

Since go-swagger regenerates all files from templates, a modified
version of the server.gotmpl was added to had the ability to modify the
sockets permissions.

Signed-off-by: André Martins <andre@cilium.io>

tgraf commented on daemon/defaults/defaults.go:
```diff
@@ -19,7 +19,7 @@ const (
 
 	// RuntimePathRights are the default access rights of the RuntimePath directory
-	RuntimePathRights = 0770
+	RuntimePathRights = 0775
```
Why do you want this to be world readable?

aanm replied:
Otherwise I have to change the directory group to belong to `ciliium` as well.

I couldn't reach the cilium socket inside without `775`

tgraf replied:
Changing the directory group ownership sounds much better to me than making the entire directory world readable.

aanm replied:
Should we change the cilium socket to `/var/run/cilium.sock` instead of `/var/run/cilium/cilium.sock`?

tgraf replied:
Yes, let's do that. That we can keep the runtime directory safe.

tgraf commented on api/v1/server/server.go:
```diff
@@ -168,6 +170,23 @@ func (s *Server) Serve() (err error) {
+					" permissions in %q: %s", common.CiliumGroupName, socketPath, err)
+			}
+		}
+
```
I think this code will be overwritten when we call, `make generate-api`, no?

aanm replied:
argh.. you're right... :( let me see if I can workaround this

aanm committed:
server: removed http listener by default

Since we don't have an option to overwrite in cilium-agent, there's no
need to expose a port for the cilium-agent to receive requests. By
default, all requests will be dealt by the cilium socket.

Signed-off-by: André Martins <andre@cilium.io>

tgraf commented on Makefile:
```diff
@@ -85,7 +85,7 @@ runtime-tests:
 generate-api:
 	swagger generate server -t api/v1 -f api/v1/openapi.yaml -a restapi \
-	    -s server -C api/v1/cilium-server.yml
+	    -s server --default-scheme=unix -C api/v1/cilium-server.yml
```
I'm fine disabling it for now until we have the options in place to support a certificate.

tgraf reviewed (approved):
Carrying an entire template file is not ideal but I don't see a better way right now.
